starfish/core/spots/FindSpots/local_max_peak_finder.py

```diff
@@ -0,0 +1,295 @@
+    stringency : int
+    min_obj_area : int
+    max_obj_area : int
+    threshold : Optional[Number]
+    measurement_type : str, {'max', 'mean'}
+        default 'max' calculates the maximum intensity inside the object
+    min_num_spots_detected : int
+        When fewer than this number of spots are detected, spot searching for higher threshold
+        values. (default = 3)
+    is_volume : bool
+        Not supported. For 3d peak detection please use TrackpyLocalMaxPeakFinder.
+        (default=False)
+    verbose : bool
+        If True, report the percentage completed (default = False) during processing
+
+    Notes
+    -----
+    http://scikit-image.org/docs/dev/api/skimage.feature.html#skimage.feature.peak_local_max
+    """
+
+    def __init__(
+        self, min_distance: int, stringency: int, min_obj_area: int, max_obj_area: int,
+        threshold: Optional[Number]=None, measurement_type: str='max',
+        min_num_spots_detected: int=3, is_volume: bool=False, verbose: bool=True
+    ) -> None:
+
+        self.min_distance = min_distance
+        self.stringency = stringency
+        self.min_obj_area = min_obj_area
+        self.max_obj_area = max_obj_area
+        self.threshold = threshold
+        self.min_num_spots_detected = min_num_spots_detected
+
+        self.measurement_function = self._get_measurement_function(measurement_type)
+
+        self.is_volume = is_volume
+        self.verbose = verbose
+
+    def _compute_num_spots_per_threshold(self, img: np.ndarray) -> Tuple[np.ndarray, List[int]]:
+        """Computes the number of detected spots for each threshold
+
+        Parameters
+        ----------
+        img : np.ndarray
+            The image in which to count spots
+
+        Returns
+        -------
+        np.ndarray :
+            thresholds
+        List[int] :
+            spot counts
+        """
+
+        # thresholds to search over
+        thresholds = np.linspace(img.min(), img.max(), num=100)
+
+        # number of spots detected at each threshold
+        spot_counts = []
+
+        # where we stop our threshold search
+        stop_threshold = None
+
+        if self.verbose and StarfishConfig().verbose:
+            threshold_iter = tqdm(thresholds)
+            print('Determining optimal threshold ...')
+        else:
+            threshold_iter = thresholds
+
+        for stop_index, threshold in enumerate(threshold_iter):
+            spots = peak_local_max(
+                img,
+                min_distance=self.min_distance,
+                threshold_abs=threshold,
+                exclude_border=False,
+                indices=True,
+                num_peaks=np.inf,
+                footprint=None,
+                labels=None
+            )
+
+            # stop spot finding when the number of detected spots falls below min_num_spots_detected
+            if len(spots) <= self.min_num_spots_detected:
+                stop_threshold = threshold
+                if self.verbose:
```

Comment:
possibly out of the scope of this PR, but this test of verbosity differs from the above one.